### ISSUE TITLE
Regarding the problem that the audio file length cannot be compressed

### ISSUE THREAD

**Issue by rezzsl** [yzGuu830/efficient-speech-codec#4]
Hello, I would like to ask, when I am compressing some audio files, some of the audio files are short, and the code requires that the time length must be a multiple of the overlap, so it cannot be processed. How can I deal with this situation to successfully compress it? Thank you.

**Comment by yzGuu830** [comment]
Hi @rezzsl, thank you for bringing this up. Our implementation actually requires that $T$ be multiple of $(\text{overlap} \cdot \text{patchsize})$ (for patchify and frame overlap), and that $L$ be multiple of $\text{hoplen}$ (to ensure lossless inverse STFT). 

For practical use of our pretrained models, I suggest either 1) truncating (with at most 20ms loss), or 2) padding (then truncating back to original length) the input audio length $L$ to be in the form $80 (4k-1), k \in \mathbb{Z}_+$. In streaming setup, you can buffer incoming audio until it satisfies the form. Alternatively, you could modify `PatchEmbed` and `PathDeEmbed` modules to include conditional padding patterns that can handle any input length.

**Comment by lixinghe1999** [comment]
need add one line to compensate
x = torch.nn.functional.pad(x, (0, 320 - x.shape[0] % 320 - 80), mode='constant', value=0)
